### PR TITLE
[BugFix] skip_done_states in SAC

### PR DESCRIPTION
**Pull request by vmoens**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #2613

### CODE REVIEW

File: torchrl/objectives/sac.py [hunk 1]
```diff
@@ -126,6 +126,10 @@ class SACLoss(LossModule):
             ``"none"`` | ``"mean"`` | ``"sum"``. ``"none"``: no reduction will be applied,
             ``"mean"``: the sum of the output will be divided by the number of
             elements in the output, ``"sum"``: the output will be summed. Default: ``"mean"``.
+        skip_done_states (bool, optional): whether the actor network should only be run on valid, non-terminating
```
Comment (matteobettini) [inline torchrl/objectives/sac.py:129]:
```suggestion
        skip_done_states (bool, optional): whether the actor network used for value computation should only be run on valid, non-terminating
```

File: torchrl/objectives/sac.py [hunk 2]
```diff
@@ -877,6 +891,10 @@ class DiscreteSACLoss(LossModule):
...
```
Comment (matteobettini) [inline torchrl/objectives/sac.py:894]: same as above

**Review by matteobettini — APPROVED** [review]
LGTM thanks!

One other option to consider to avoid changing the input shape in case the user sets this flag is to ask what to use to pad the non-terminated obs

**Comment by vmoens** [comment]
> LGTM thanks!
> 
> One other option to consider to avoid changing the input shape in case the user sets this flag is to ask what to use to pad the non-terminated obs

See [this comment](https://github.com/pytorch/rl/issues/2612#issuecomment-2504137606)

**Comment by matteobettini** [comment]
> > LGTM thanks!
> > One other option to consider to avoid changing the input shape in case the user sets this flag is to ask what to use to pad the non-terminated obs
> 
> See [this comment](https://github.com/pytorch/rl/issues/2612#issuecomment-2504137606)

Yes I saw. I was referring exactly to that. Maybe there are users who have such issue but also need an input with the same shape (cholesky expects a matirx for isntance). in that case they might know what works for them (maybe nana and 0 no but 1 yes)

Just an idea, we don't need to do it

**Comment by vmoens** [comment]
> Yes I saw. I was referring exactly to that. Maybe there are users who have such issue but also need an input with the same shape (cholesky expects a matrix for isntance). in that case they might know what works for them (maybe nana and 0 no but 1 yes)

with cholesky (if we want to take that example), any fixed number filling the matrix will fail. Padding is simply not a solution unfortunately.

**Comment by matteobettini** [comment]
> with cholesky (if we want to take that example), any fixed number filling the matrix will fail. Padding is simply not a solution unfortunately.

got it. but neither is masking if i understand right

**Comment by matteobettini** [comment]
ok got it

**Pull request merged**